### PR TITLE
Fixing bazel build //... and Follow up to #5320

### PR DESCRIPTION
**Pull request by sebastienvas**
#5320 was not touched since 15 days, so I took a look at it too:

What was required:
Adds overrides for k8s.io/api and machinery
dep prune
Remove the BUILD files in vendor before running hack/update-bazel.sh
Adds overrides for gopkg.in/robfig/cron.v2. deps does not seem to be able to find use v2. So I had to edit the Gopkg.lock myself and run ensure after.

### CODE REVIEW

**Comment by cblecker** [comment]
interesting.. https://github.com/kubernetes/test-infra/pull/5320#issuecomment-343341521 didn't hit. hmm.. I'm guessing because now there is a `hack/` dir in k8s.io/test-infra?

I'm going to test something..

**Comment by sebastienvas** [comment]
@cblecker this is because I manually removed the BUILD files from vendors before running hack/update-bazel.sh. I also had to dep prune.

**Comment by cblecker** [comment]
I don't think that's going to work. The next time you run `dep ensure` it's going to blow away those manual changes (along with all of `vendor/`).

**Comment by BenTheElder** [comment]
> @cblecker this is because I manually removed the BUILD files from vendors before running hack/update-bazel.sh. I also had to dep prune.

> I don't think that's going to work. The next time you run dep ensure it's going to blow away those manual changes (along with all of vendor/).

We could just be super hacky and add these to `hack/update-bazel.sh` 🤷‍♂️

**Comment by sebastienvas** [comment]
dep ensure works just fine as I added override in Gopkg.toml, which is way you are supposed to do it. I just tried it works. dep prune is documented here already
https://github.com/kubernetes/test-infra/blob/master/docs/dep.md
While removing BUILD files from vendor is fine since gazelle recreates them, I am not confident calling dep prune. I ll let you guys deal with this. I don't think there is a working recipe just yet

**Comment by sebastienvas** [comment]
Just added the delete BUILD to hack/update-bazel.sh

**Comment by cblecker** [comment]
I did a similar, but smaller scoped thing in https://github.com/kubernetes/test-infra/pull/5320. This PR is 2 commits and changes over 200k LOC. Difficult to review the commits aren't split up.

**Comment by sebastienvas** [comment]
Well I do that, because you are going to ask me to squash them anyway. I am just trying to speed this up. If you remove the vendor dir the changes are pretty easy to review.

**Comment by cblecker** [comment]
Squash usually happens after import. I'm also curious.. it looks like you're updating a bunch of deps to master.. is there a reason for that?

**Comment by sebastienvas** [comment]
Yes https://github.com/kubernetes/contrib/blob/master/peer-finder/peer-finder.go used to need k8s.io/kubernetes. So I forced it to update to master.

File: Gopkg.lock [hunk 1]
```diff
@@ -2,79 +2,87 @@
 
 
 [[projects]]
-  name = "bitbucket.org/ww/goautoneg"
-  packages = ["."]
-  revision = "'75cd24fc2f2c2a2088577d12123ddee5f54e0675'"
-
-[[projects]]
+  branch = "master"
```
Comment (cblecker) [inline Gopkg.lock]: But this also upgrades `github.com/NYTimes/gziphandler` to master

Comment (sebastienvas) [reply Gopkg.lock]: dep did that not me, probably because those projects don't have releases or any other branches.

Comment (cblecker) [reply Gopkg.lock]: I'm guessing you ran a `dep ensure -update` without specifying a package, so it attempted to upgrade all packages (not something I'm sure we want to do)

**Comment by BenTheElder** [comment]
- I think normally we don't squash all commits, just noisy ones so that it's easy to look through the commit history. It is pretty hard to see what a single commit does for something this big even after merge.
- I'm also not sure if we should update all dependencies, but maybe we should? I think we should discuss this somewhere if not now, after these PRs are resolved. Hopefully vendor/ isn't sitting on a pile of vulnerabilities...
  - Somewhat orthogonal but I still think we should use something like NGINX for gzip instead of the go services anyhow ...

The basic idea is LGTM. Both PRs need a rebase though.

**Pull request closed without merging**